### PR TITLE
SCA: simplify some ifs in favour of null coalescing operator

### PR DESCRIPTION
**Pull request by kalessil**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Some of ifs were replaced with null coalescing operator in order to simplify control flow.

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -54,12 +54,12 @@ public function copy($originFile, $targetFile, $overwriteNewerFiles = false)
 
         if ($doCopy) {
             // https://bugs.php.net/bug.php?id=64634
-            if (false === $source = @fopen($originFile, 'r')) {
+            if (false === $source = @fopen($originFile, 'rb')) {
```
Comment (Kocal) [inline src/Symfony/Component/Filesystem/Filesystem.php]: This should be reverted no? (same for the other same changes)

Comment (kalessil) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Yep, but CS bot requires the changes.

Comment (Kocal) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Oh, okay :man_shrugging:

Comment (nicolas-grekas) [reply src/Symfony/Component/Filesystem/Filesystem.php]: can you revert? that's a bug of the cs bot (already reported on cs-fixer)

Comment (kalessil) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Reverting =)

Comment (kalessil) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Done!

**Review by nicolas-grekas — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @kalessil.

**Comment by nicolas-grekas** [comment]
@kalessil for your next PR, could you please use a single github users to make commit? Or at least squash the commits? We prefer squashing PR sub-history, but we don't so when several author are involved. I inadvertently merged the 4 commits here while they should have been squashed.

**Comment by kalessil** [comment]
@nicolas-grekas, oh sorry - I've overseen this point. Will do for sure.

**Pull request merged**